packages/typeahead/src/react/index.js

```diff
@@ -0,0 +1,250 @@
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import Icon from '@pluralsight/ps-design-system-icon/react.js'
+import { BelowLeft } from '@pluralsight/ps-design-system-position/react.js'
+import { elementOfType } from '@pluralsight/ps-design-system-prop-types'
+import TextInput from '@pluralsight/ps-design-system-textinput/react.js'
+
+import * as vars from '../vars/index.js'
+
+import useOnDocumentClick from './use-on-document-click.js'
+import usePortal from './use-portal.js'
+import { combineFns, omit, pick } from './utils.js'
+
+const TEXT_INPUT_PROPS = [
+  'appearance',
+  'disabled',
+  'error',
+  'label',
+  'name',
+  'onChange',
+  'placeholder',
+  'subLabel',
+  'value'
+]
+const Typeahead = React.forwardRef((props, forwardedRef) => {
+  const { children, filterFn, onChange, value } = props
+
+  const portal = usePortal()
+  const containerRef = React.useRef()
+
+  const fieldRef = React.useRef()
+  const inputRef = React.useRef()
+  React.useImperativeHandle(forwardedRef, () => inputRef.current)
+
+  const [controlled] = React.useState(typeof value !== 'undefined')
+  const [open, setOpen] = React.useState(false)
+
+  const [innerValue, setInnerValue] = React.useState(value || '')
+  const [searchTerm, setSearchTerm] = React.useState('')
+
+  React.useEffect(
+    function updateControlledValue() {
+      if (controlled) setInnerValue(value)
+    },
+    [controlled, value]
+  )
+
+  React.useEffect(
+    function clearSearchTermOnClose() {
+      if (!open) setSearchTerm('')
+    },
+    [open]
+  )
+
+  useOnDocumentClick(evt => {
+    const isInnerClick =
+      !containerRef.current ||
+      containerRef.current.contains(evt.target) ||
+      portal.current.contains(evt.target)
+
+    if (isInnerClick) return
+
+    setOpen(false)
+  })
+
+  const suggestions = React.useMemo(() => {
+    const childArray = Children.toArray(children)
+    return childArray.map((child, index) => ({
+      index,
+      label: getSuggestionLabel(child),
+      value: getSuggestionValue(child)
+    }))
+  }, [children])
+
+  const filteredSuggestions = React.useMemo(
+    () => filterFn(searchTerm, suggestions),
+    [filterFn, searchTerm, suggestions]
+  )
+
+  const handleChange = (evt, nextValue) => {
+    setSearchTerm(nextValue)
+
+    if (!controlled) setInnerValue(nextValue)
+    if (onChange) onChange(evt, nextValue)
+  }
+
+  const handleInputFocus = combineFns(evt => {
+    setOpen(true)
+  }, props.onFocus)
+
+  const handleInputChange = evt => {
+    const nextValue = evt.target.value
+
+    handleChange(evt, nextValue)
+  }
+
+  const handleSuggestionMenuChange = (evt, nextValue) => {
+    setOpen(false)
+    handleChange(evt, nextValue)
+  }
+
+  return (
+    <div
+      {...filterReactProps(omit(props, TEXT_INPUT_PROPS))}
+      ref={containerRef}
+    >
+      <BelowLeft
+        inNode={portal.current}
+        when={open}
+        show={
+          <SuggestionsMenu
```

Comment:
No idea what might be causing this, but seeing weird scroll issues.  Happens in Chrome and FF.  Happens with scroll wheel on mouse but not on the Mac trackpad.

Would you mind seeing if you or another person have the same mouse wheel problem?  If it's just me, maybe something unrelated to the code.

FF:
![ff](https://user-images.githubusercontent.com/552044/65986137-8318ef00-e440-11e9-9954-b5268cb4fbcf.gif)

Chrome:
![chrome](https://user-images.githubusercontent.com/552044/65986138-8318ef00-e440-11e9-9a39-74bc4104a245.gif)